### DISCUSSION TITLE
https://angular.love/angular-facade-pattern

### DISCUSSION THREAD

**Discussion by giscus** [HouseOfAngular/angular-love#337] [General]
# https://angular.love/angular-facade-pattern

Angular.love - a place for all Angular enthusiasts created to inspire and educate.

https://angular.love/angular-facade-pattern

**Comment by dainius-s** [comment]
The article theoretical part is fine but actual example is wrong that facade doesn't make much sense.

AuthService - should only be responsible for authentication. Meaning if you want to ensure user is authenticated the you use route guard instead. 

In conjunction with AuthGuard one would normally use Global 401 error interceptor to log user out automatically.

UserDetails - should be resolved at the router level once AuthGuard resolved as true.


This means your example doesn't have real use case. I would advise creating a much more realistic use case instead.

I often see over-engineering in Angular while overlooking simple solutions/tools that Angular offer.

**Reply by mateusz-dobrowolski-va** [reply]
Thx for feedback!
After some time I also think that the example is not entirely appropriate and can be confusing. I will make corrections soon, changing the example to a more life-like one

**Comment by pawel-twardziak** [comment]
Interesting contents, Thanks @mateusz :)

**Reply by pawel-twardziak** [reply]
Damn, should be @mateusz-dobrowolski-va , sorry for the mistake :P

**Reply by mateusz** [reply]
_says in a deep voice as if coming from the deepest crypts_ YOU SUMMONED ME?

**Reply by pawel-twardziak** [reply]
sorry, it was by mistake :D And I cannot edit it hehe
